uktrade/data-hub-frontend
# Add opportunity edit functionality - name only

## Description of change

This PR is the first in a series of PRs towards adding edit functionality to a UK Opportunity. This particular PR allows for an edit to be made for the name only. If this work is accepted, then the next PR will work on building up the Opportunity Details form, and the Opportunity requirements form. 

This PR contains tech debt. It's my current understanding that the Form component has a bug in it, which means that it doesn't actually wait for the Task it holds to finish, before doing the next thing which can result in things not updating as quickly as we would want. My PR uses this broken form component, as there's no agreed alternative. 

## Test instructions

Go to investments > UK opportunities and click on an opportunity. Click on the opportunity details toggle, and click the Edit button at the bottom of the section. Fill out a new name and click Submit. You should be redirected back to the section with the summary table in it, showing your new name.

## Screenshots
### Before

_Add a screenshot_

### After

_Add a screenshot_

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
